umbra-js/src/classes/Umbra.ts

```diff
@@ -149,7 +149,7 @@ export class Umbra {
    */
   constructor(readonly provider: EthersProvider, chainConfig: ChainConfig | number) {
     this.chainConfig = parseChainConfig(chainConfig);
-    this.umbraContract = new Contract(this.chainConfig.umbraAddress, UMBRA_ABI, provider) as UmbraContract;
+    this.umbraContract = new Contract(this.chainConfig.umbraAddress, UMBRA_ABI, provider) as unknown as UmbraContract;
```

Comment:
I think we can remove these unknowns. Instead use the UmbraFactory from typechain to get a new contract. It should look like the below

```
this.umbraContract = UmbraFactory.connect(this.chainConfig.umbraAddress, provider);
```

Reply:
Like this: https://github.com/ScopeLift/umbra-protocol/pull/577/commits/92de58d332ad2e4814a9569b068bdca6524da124 ?

Reply:
Yes